data_model/src/events/time.rs

```diff
@@ -56,57 +62,61 @@ impl EventFilter {
-                let start = schedule.start + Self::multiply_duration_by_u128(period, k);
-                let range = Range::from(*interval);
-                (0..)
-                    .map(|i| start + period * i)
-                    .skip_while(|time| *time < interval.since)
-                    .take_while(|time| range.contains(time))
-                    .count()
-                    .try_into()
-                    .expect(
-                        "Overflow. The schedule is too frequent relative to the interval length",
-                    )
-            },
+    fn mintable(&self) -> bool {
+        !matches!(
+            self.0,
+            ExecutionTime::Schedule(Schedule { period: None, .. })
         )
     }
+}
 
-    /// Multiply `duration` by `n`
-    ///
-    /// Usage of this function allows to operate with much longer time *intervals*
-    /// with much less *periods* than just `impl Mul<u32> for Duration` does
-    ///
-    /// # Panics
-    /// Panics if resulting number in seconds can't be represented as `u64`
-    #[allow(clippy::expect_used, clippy::integer_division)]
-    fn multiply_duration_by_u128(duration: Duration, n: u128) -> Duration {
-        if let Ok(n) = u32::try_from(n) {
-            return duration * n;
-        }
+/// Count something with the `schedule` within the `interval`
+#[allow(clippy::expect_used)]
+fn count_matches_in_interval(schedule: &Schedule, interval: &Interval) -> u32 {
+    schedule.period.map_or_else(
+        || u32::from(Range::from(*interval).contains(&schedule.start)),
+        |period| {
+            #[allow(clippy::integer_division)]
+            let k = interval.since.saturating_sub(schedule.start).as_millis() / period.as_millis();
+            let start = schedule.start + multiply_duration_by_u128(period, k);
+            let range = Range::from(*interval);
+            (0..)
+                .map(|i| start + period * i)
+                .skip_while(|time| *time < interval.since)
+                .take_while(|time| range.contains(time))
+                .count()
+                .try_into()
+                .expect("Overflow. The schedule is too frequent relative to the interval length")
+        },
+    )
+}
 
-        let new_ms = duration.as_millis() * n;
-        if let Ok(ms) = u64::try_from(new_ms) {
-            return Duration::from_millis(ms);
-        }
+/// Multiply `duration` by `n`
+///
+/// Usage of this function allows to operate with much longer time *intervals*
+/// with much less *periods* than just `impl Mul<u32> for Duration` does
+///
+/// # Panics
+/// Panics if resulting number in seconds can't be represented as `u64`
+#[allow(clippy::expect_used, clippy::integer_division)]
+fn multiply_duration_by_u128(duration: Duration, n: u128) -> Duration {
+    if let Ok(n) = u32::try_from(n) {
+        return duration * n;
+    }
 
-        let new_secs = u64::try_from(new_ms / 1000)
```

Comment:
This is an important change. Good catch.

Reply:
What he means is that you "caught" a bug from bothering us is the future.

Reply:
Sorry, the expression "Good catch" expands to "You caught a subtle bug that isn't obvious and requires paying attention".

Reply:
Yeah, I know it :)
My question is that I can't see a bug here